src/poliastro/maneuver.py

```diff
@@ -41,12 +41,13 @@ def __init__(self, *args):
         self.impulses = args
         # HACK: Change API or validation code
         _dts, _dvs = zip(*args)
-        self._dts, self._dvs = self._initialize(
-            [(_dt * u.one).value for _dt in _dts] * (_dts[0] * u.one).unit,
-            [(_dv * u.one).value for _dv in _dvs] * (_dvs[0] * u.one).unit,
-        )
+
         try:
-            if not all(len(dv) == 3 for dv in self._dvs):
+            self._dts, self._dvs = self._initialize(
+                [(_dt * u.one).value for _dt in _dts] * (_dts[0] * u.one).unit,
+                [(_dv * u.one).value for _dv in _dvs] * (_dvs[0] * u.one).unit,
+            )
+            if not all(len(dv) == 3 for dv in _dvs):
```

Comment:
If this `TypeError` ever raised? I see it's not covered by the tests. Maybe it's dead code and we should remove it? Otherwise, let's use this opportunity to add a test

Reply:
The test that was added checks for the line:

```python
if not all(len(dv) == 3 for dv in _dvs):
```

and since

```python
self._dts, self._dvs = self._initialize(
                [(_dt * u.one).value for _dt in _dts] * (_dts[0] * u.one).unit,
                [(_dv * u.one).value for _dv in _dvs] * (_dvs[0] * u.one).unit,
            )
```
seems to raises an error only when `len(dv) != 3`, the error message might handle both. What do you think?

Reply:
Then let's remove that code, looks like that case is covered already

Reply:
Sorry, but now it looks like the test `test_maneuver_raises_error_if_dvs_are_not_vectors` raises an error. This is because removing the `if not all(len(dv) == 3 for dv in _dvs):` check would also allow delta-v's that are not 3 dimensional, which might not be favourable.

Reply:
Maybe we could keep the condition:

```py
if not all(len(dv) == 3 for dv in _dvs):
```
unchanged due to the above reason. Could you let me know what do you think @astrojuanlu ?

Reply:
Fair enough, if there is a test checking for it, it's not dead code then. Let's keep it.